Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -549,7 +563,9 @@ bool MediaSource::hasFutureTime()
     if (isClosed())
         return false;
 
-    return m_private->hasFutureTime(currentTime(), m_private->timeIsProgressing() ? MediaTime::zeroTime() : MediaSourcePrivate::futureDataThreshold());
+    Ref msp = protectedPrivate().releaseNonNull();
```

Comment:
Ditto

Reply:
same. it's very explicit here that msp is a Ref and is non-null.